Remove golden of deprecated `flag_output_set_profile` test

kailun-qin opened the pull request:
The `flag_output_set_profile` manifest generate flags test was refactored and
moved into `TestManifestGenerateFlagsMinimalProfile` by
https://github.com/istio/istio/pull/25405. However, the related golden file was
introduced accidentally by https://github.com/istio/istio/pull/25263.

The golden file is deprecated, not used by tests and contains stale information
(annotations etc.), which should be removed.

Signed-off-by: Kailun Qin <kailun.qin@intel.com>

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure


Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

irisdingbj reviewed (approved):
Thanks!

istio-testing commented:
@kailun-qin: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-security-multicluster-tests_istio | 5e954f1c3ec1db1a690ac598e68679a50be3b9bd | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/32733/integ-security-multicluster-tests_istio/1390915987439620096) | `/test integ-security-multicluster-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

kailun-qin commented:
/retest

kailun-qin committed:
Remove golden of deprecated `flag_output_set_profile` test

The `flag_output_set_profile` manifest generate flags test was refactored and
moved into `TestManifestGenerateFlagsMinimalProfile` by
https://github.com/istio/istio/pull/25405. However, the related golden file was
introduced accidentally by https://github.com/istio/istio/pull/25263.

The golden file is deprecated, not used by tests and contains stale information
(annotations etc.), which should be removed.

Signed-off-by: Kailun Qin <kailun.qin@intel.com>
